cli/auth.go

```diff
@@ -83,6 +83,11 @@ type tokenResponse struct {
 	TokenType    string `json:"token_type"`
 }
 
+type authMethod interface {
+	dialOpts() rpc.DialOption
+	prettyPrint() string
```

Comment:
I think you should embed `fmt.Stringer` and implement `String` for both `token` and `apiKey` instead of `prettyPrint`. You can just use the `%s` print directive in print statements to automatically invoke the method, too.